target/linux/ath79/image/generic.mk

```diff
@@ -355,6 +355,31 @@ define Device/dlink_dir-859-a1
 endef
 TARGET_DEVICES += dlink_dir-859-a1
 
+define Device/dlink_dir-842-c2
+  KERNEL := kernel-bin | append-dtb | relocate-kernel | lzma
+  KERNEL_INITRAMFS := $$(KERNEL) | seama
+  IMAGES += factory.bin
+  SEAMA_MTDBLOCK := 5
+  # 64 bytes offset:
+  # - 28 bytes seama_header
+  # - 36 bytes of META data (4-bytes aligned)
+  IMAGE/default := append-kernel | uImage lzma | pad-offset $$$$(BLOCKSIZE) 64 | append-rootfs
+  IMAGE/sysupgrade.bin := \
+	$$(IMAGE/default) | seama | pad-rootfs | append-metadata | check-size $$$$(IMAGE_SIZE)
+  IMAGE/factory.bin := \
+	$$(IMAGE/default) | pad-rootfs -x 64 | seama | seama-seal | check-size $$$$(IMAGE_SIZE) 
+  ATH_SOC := qca9563
+  DEVICE_VENDOR := D-Link
+  DEVICE_MODEL := DIR-842
+  DEVICE_VARIANT := C2
+  IMAGE_SIZE := 15680k
+  DEVICE_PACKAGES :=  kmod-usb-core kmod-usb2 kmod-ath10k-ct ath10k-firmware-qca9888-ct
+  SEAMA_SIGNATURE := wrgac65_dlink.2015_dir842
+  SUPPORTED_DEVICES += dir-842-c2
```

Comment:
Has this been supported in ar71xx? If not, remove this line.

Reply:
I think SUPPORTED_DEVICES is here for every devices?

Reply:
No, this is for upgrade compatibility with ar71xx.

Reply:
> No, this is for upgrade compatibility with ar71xx.

Thanks! I remove it

Reply:
That's why you do "+=". The string used for ath79 is calculated automatically based on the Device node name, and you are just adding a different one.